cachito/workers/tasks/general.py

```diff
@@ -18,10 +18,48 @@
     "failed_request_callback",
     "fetch_app_source",
     "set_request_state",
+    "get_request_state",
+    "check_state",
 ]
 log = logging.getLogger(__name__)
 
 
+def get_request_state(request_id):
+    """
+    Get the state of the request.
+
+    :param int request_id: the Cachito request ID this is for
+    """
+    from cachito.workers.requests import requests_auth_session
+
+    config = get_worker_config()
+    request_url = f'{config.cachito_api_url.rstrip("/")}/requests/{request_id}'
+
+    rv = requests_auth_session.get(request_url)
+
+    try:
+        rv.raise_for_status()
+    except requests.RequestException:
+        msg = f"An error occured when getting the state of request {request_id}"
+        log.exception(msg)
+        raise CachitoError(msg)
+
+    return rv.json()["state"]
+
+
+def check_state(request_id):
```

Comment:
For current implementation, suggest to rename this function name to `is_request_in_progress`.